### PR TITLE
can produce an array of comments for a particular article and error w…

### PR DESCRIPTION
**Pull request by NRMSMITH**
…hen an article doesnt exist - should also produce an empty array for articles that exist but no comments

### CODE REVIEW

File: __tests__/api.test.js [hunk 1]
```diff
@@ -294,3 +294,42 @@ describe('GET /api/articles', () => {
...
+describe('GET /api/articles/:article_id/comments', () => {
+    test('200: should respond with an array of comments for a particular article', () => {
+        const article_id = 1
+        return request(app)
+        .get(`/api/articles/${article_id}/comments`)
+        .expect(200)
+        .then(({body}) => {
+            expect(Array.isArray(body)).toBe(true);
+            expect(body.length === 11).toBe(true);
+            body.forEach((comment) => {
```
Comment (popatre) [inline __tests__/api.test.js:306]: These need to be returned on a key of `comments`

File: __tests__/api.test.js [hunk 2]
```diff
@@ -294,3 +294,42 @@ describe('GET /api/articles', () => {
...
+                })
+            })
+        })
+    })
+    test('200: should respond with an empty array when the article exists but no comments are posted', () => {
+        return request(app)
+        .get(`/api/articles/2/comments`)
+        .expect(200)
+        .then(({body}) => {
+            expect(Array.isArray(body)).toBe(true);
```
Comment (popatre) [inline __tests__/api.test.js:322]: should have comment key

File: __tests__/api.test.js [hunk 3]
```diff
@@ -294,3 +294,42 @@ describe('GET /api/articles', () => {
...
+    })
+    test('404: should respond with an error saying not found when the article id does not exist', () => {
+        return request(app)
+        .get(`/api/articles/5000/comments`)
+        .expect(404)
+        .then(({body}) => {
+            expect(body.msg).toBe('Article ID does not exist');
+        })
+    });
+})
```
Comment (popatre) [inline __tests__/api.test.js:335]: Needs another test

- [ ] 400 invalid article id

File: api/controllers/comments.controllers.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const {selectCommentsById} = require('../models/comments.models')
+
+exports.getCommentsById = (req, res, next) => {
+    const {article_id} = req.params;
+    selectCommentsById(article_id).then((comments) => {
+        return res.status(200).send(comments)
```
Comment (popatre) [inline api/controllers/comments.controllers.js:6]: Need to return an object

**Review by popatre — CHANGES_REQUESTED** [review]

**Review by xViolaine — APPROVED** [review]

**Pull request merged**